src/Widgets/FirmwareUpdateRow.vala

```diff
@@ -0,0 +1,176 @@
+            if (device.is (DeviceFlag.NEEDS_REBOOT)) {
+                show_reboot_dialog ();
+            } else if (device.is (DeviceFlag.NEEDS_SHUTDOWN)) {
+                show_shutdown_dialog ();
+            }
+        }
+    }
+
+    private void show_details_dialog (Details details) {
+        var message_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("Manual steps required"),
+            details.caption,
+            "application-x-firmware",
+            Gtk.ButtonsType.NONE
+        );
+        message_dialog.transient_for = (Gtk.Window) get_toplevel ();
+
+        var suggested_button = new Gtk.Button.with_label (_("Continue"));
+        suggested_button.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
+        message_dialog.add_action_widget (suggested_button, Gtk.ResponseType.OK);
+
+        if (details.image != null) {
+            var custom_widget = new Gtk.Image.from_file (details.image);
+            message_dialog.custom_bin.add (custom_widget);
+        }
+
+        message_dialog.badge_icon = new ThemedIcon ("dialog-information");
+        message_dialog.show_all ();
+        message_dialog.run ();
+        message_dialog.destroy ();
+    }
+
+    private void show_reboot_dialog () {
+        var message_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("An update requires a reboot to complete"),
+            _("This will close all open applications and restart this device."),
+            "application-x-firmware",
+            Gtk.ButtonsType.CANCEL
+        );
+        message_dialog.transient_for = (Gtk.Window) get_toplevel ();
+
+        var suggested_button = new Gtk.Button.with_label (_("Reboot"));
+        suggested_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+        message_dialog.add_action_widget (suggested_button, Gtk.ResponseType.OK);
+
+        message_dialog.badge_icon = new ThemedIcon ("dialog-information");
+        message_dialog.show_all ();
+        if (message_dialog.run () == Gtk.ResponseType.OK) {
+            LoginManager.get_instance ().reboot ();
+        }
+
+        message_dialog.destroy ();
+    }
+
+    private void show_shutdown_dialog () {
+        var message_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("An update requires the system to shutdown to complete"),
+            _("This will close all open applications and turn off this device."),
+            "application-x-firmware",
+            Gtk.ButtonsType.CANCEL
+        );
+        message_dialog.transient_for = (Gtk.Window) get_toplevel ();
+
+        var suggested_button = new Gtk.Button.with_label (_("Shutdown"));
+        suggested_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+        message_dialog.add_action_widget (suggested_button, Gtk.ResponseType.OK);
+
+        message_dialog.badge_icon = new ThemedIcon ("dialog-information");
```

Comment:
```suggestion
        message_dialog.badge_icon = new ThemedIcon ("system-shutdown");
```

Reply:
I have also changed it for reboot based on your suggestion